service/policy/db/namespaces.go

```diff
@@ -386,3 +398,212 @@ func (c PolicyDBClient) RemovePublicKeyFromNamespace(ctx context.Context, k *nam
 		KeyId:       k.GetKeyId(),
 	}, nil
 }
+
+// validateRootCertificate validates that the PEM string is a valid PEM-encoded root certificate
+func validateRootCertificate(pemStr string) error {
```

Comment:
I don't see any temporal checks on the certificates here.  Even if we don't currently expire the certs or set Before times, adding the checks proactively could help protect a future change where certificate rotation is introduced.

Reply:
Temporal validation checking both NotBefore and NotAfter timestamps